pkg/block/indexheader/lazy_binary_reader.go

```diff
@@ -99,30 +98,34 @@ func NewLazyBinaryReader(
 	metrics *LazyBinaryReaderMetrics,
 	onClosed func(*LazyBinaryReader),
 ) (*LazyBinaryReader, error) {
-	filepath := filepath.Join(dir, id.String(), block.IndexHeaderFilename)
+	indexHeaderFile := ""
+	if dir != "" {
+		indexHeaderFile = filepath.Join(dir, id.String(), block.IndexHeaderFilename)
+	}
 
-	// If the index-header doesn't exist we should download it.
-	if _, err := os.Stat(filepath); err != nil {
-		if !os.IsNotExist(err) {
-			return nil, errors.Wrap(err, "read index header")
-		}
+	if indexHeaderFile != "" {
```

Comment:
Alternative: We could check directly for empty `dir` here and build the `filepath` inside the loop instead of doing the declarations and checks above.

Reply:
Done